charmcraft/linters.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Analyze and lint charm structures and files."""
+
+import os
+import pathlib
+import shlex
+from collections import namedtuple
+
+CheckType = namedtuple("CheckType", "trait warning error")(
+    trait="trait", warning="warning", error="error"
+)
+
+
+class Language:
+    """Check the language used to write the charm.
+
+    Currently only Python is detected, if the following checks are true:
+
+    - the charm has a text dispatch with a python call
+    - the charm has a `.py` entry point
+    - the entry point file is executable
```

Comment:
If directly executable, it should have an interpreter that should perhaps be validated?

Reply:
We assume that if it's a .py you put the proper shebang line inside. What would be a "valid interpreter"? `/usr/local/bin/python`? `/bin/python3.7`? `/usr/bin/pypy`? 

This fits more in the "ensure it will be run under Juju environment" than "detect if it's Python", probably we could add it in the future if needed, as another linter.

Reply:
Fair enough :+1: